htlcswitch/mailbox.go

```diff
@@ -548,6 +548,8 @@ func (m *memoryMailBox) mailCourier(cType courierType) {
 				m.pktCond.L.Unlock()
 
 			case <-deadline:
+				log.Debugf("Expiring add htlc with "+
```

Comment:
Ah, so because `deadline` will only ever send when `add` was initialized as well, `add` can never be `nil` here?

Reply:
Yeah I think so. We'll only have the `deadline` when `nextAdd` is not `nil`, and fetch the `add` from `add = nextAdd.pkt `, (unless the `pkt` is `nil`, then we'd have a bigger problem as the `FailAdd` would panic).